### DISCUSSION TITLE
Sync state of two switches on two different HAA devices?

### DISCUSSION THREAD

**Discussion by RunnerMojo** [RavenSystem/esp-homekit-devices#1954] [General]
I have a set of toggle switches on one HAA device where I want to keep state in sync with other HAA controlled devices.  Similar to the way a wired set of 3-way switches work.  This works if I setup four automations in Homekit that turn the opposite switch on or off based a given switch.  So this requires two automations for each side for both on an off.    And this can quickly add up for many switches.

This does work but it does not seem elegant. Is this the best or only solution or is there a better way to do this with HAA network requests or something I am missing in Homekit or Homekit Automations?
Thanks!

**Comment by paulzhdanov** [comment]
Probably you're looking for this https://github.com/RavenSystem/esp-homekit-devices/wiki/General-Configuration#Enable-Unsecure-HomeKit-rest-API

**Reply by RunnerMojo** [reply]
My application is a scene controller. So there are momentary buttons that appear as toggle switches that are used to control other accessories or groups of accessories. I was aware of this feature in HAA, but didn't want to enable insecure Rest API. I guess there's no other way to do this in HomeKit. This is pretty common with other home automation systems, so hopefully Apple will add.
